zerver/lib/actions.py

```diff
@@ -4645,6 +4656,20 @@ def do_update_message(user_profile: UserProfile, message: Message,
         if new_stream is not None:
             assert stream_being_edited is not None
 
+            if subs_gaining_usermessages:
+                ums_to_create = []
+                for msg in changed_messages:
+                    for user_profile_id in subs_gaining_usermessages:
+                        # TODO: Add mention data to UserMessage.
+                        um = UserMessageLite(
+                            user_profile_id=user_profile_id,
+                            message_id=msg.id,
+                            flags=0,
```

Comment:
If we're setting read, we should just do it on this line.

I think there's an argument that we shouldn't set `read` in some sense -- these messages ARE new to the user.  

The problem with that argument is that we don't currently have client-side API handler support for adding new unread messages that are not new?  So there's a bit of a rabbit hole here.  Maybe worth a comment an opening an issue to investigate if we can mark these as unread.

I guess maybe more importantly -- setting these messages as unread is kinda annoying and takes away the ability for the person doing the migration to decide whether to draw attention to it via the notification checkboxes, or not.  So I think the right decision is to go with `flags=UserMessage.flags.read`.

Reply:
I don't think we should mark them as read, it will only confuse users, especially in case old messages are moved.

Reply:
Do you mean "not as read" or "not as unread"?  I think the correct answer is marking them as read, and that sounds like what you're proposing from the second half of the sentence, so we should probably do that (I think you may have accidentally said the opposite?)

Reply:
Yeah, I meant we should mark them as read. 😅